src/Context/UserInfo.tsx

```diff
@@ -1,16 +1,37 @@
-import { createContext, useContext, useState } from "react";
+import { createContext, useContext, useReducer } from "react";
 
 interface Person {
   name: string;
   lastName: string;
   isLoggedIn: boolean;
-  cars: [{ brand: string; color: string }, { brand: string; color: string }];
+  cars: Car[];
   toggleLogin?: () => void;
+  addCar?: () => void;
```

Comment:
I'd suggest to separate this, because you are mixing you state interface with the functions you want to have, but you are not changing this functions the only thing that changes is the properties of the Person